core/shared/src/main/scala/fs2/internal/CompileScope.scala

```diff
@@ -222,20 +224,24 @@ private[fs2] final class CompileScope[F[_]] private (
     * finalized after this scope is closed, but they will get finalized shortly after. See [[Resource]] for
     * more details.
     */
-  def close(ec: ExitCase[Throwable]): F[Either[Throwable, Unit]] =
+  def close(ec: ExitCase[Throwable]): F[Either[Throwable, Unit]] = {
+    val selfClean: F[Unit] = F.productR(
```

Comment:
This was nested inside the flatMap continuation, but it happens to be independent of the variables bound in that flatMap.